File: sdk/java/src/main/java/com/lightswitch/domain/Variation.java
package com.lightswitch.domain;

public class Variation {

	private int order;
	private String value;
	private int portion;
	private String description;

	public Variation(int order, String value, int portion, String description) {
		this.order = order;
		this.value = value;
		this.portion = portion;
		this.description = description;
	}

	public int getPortion() {
		return portion;
	}

	public String getValue() {
		return value;
	}
}

File: backend/core-service/src/main/kotlin/com/lightswitch/core/domain/flag/repository/entity/Variation.kt
package com.lightswitch.core.domain.flag.repository.entity

import com.lightswitch.core.common.entity.BaseEntity
import com.lightswitch.core.domain.history.dto.PreVariation
import jakarta.persistence.*

@Entity(name = "variation")
class Variation(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val variationId: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "flag_id")
    val flag: Flag,

    var value: String,
    var portion: Int,
    var description: String,

    val defaultFlag: Boolean = false,
) : BaseEntity() {

    fun toPrevious(): PreVariation {
        return PreVariation(this)
    }
}